### PR TITLE
Validate build directory gamelift

### PR DESCRIPTION
**Pull request by kyleknap**
The build directory needs to exist and be nonempty to upload builds to GameLift.

cc @jamesls @JordonPhillips

### CODE REVIEW

File: awscli/customizations/gamelift/uploadbuild.py [hunk 1]
```diff
@@ -102,6 +111,13 @@ def zip_directory(zipfile_name, source_root):
                     zf.write(full_path, relative_path)
 
 
+def validate_directory(source_root):
+    for path, dirs, files in os.walk(source_root):
```
Comment (jamesls) [inline awscli/customizations/gamelift/uploadbuild.py]: Is there a reason we can't just use `return os.path.isdir(source_root) and os.listdir(source_root)`?

**Comment by jamesls** [comment]
:shipit:  Had one small comment, otherwise looks good.

File: awscli/customizations/gamelift/uploadbuild.py (hunk 1 above)
Comment (kyleknap) [inline awscli/customizations/gamelift/uploadbuild.py]: Yeah I will add it.

**Comment by kyleknap** [comment]
I added a comment. Merging.

**Pull request merged**